programs/zstdcli.c

```diff
@@ -1013,8 +1013,16 @@ int main(int argCount, const char* argv[])
                 }
 #endif
                 if (longCommandWArg(&argument, "--threads")) { NEXT_UINT32(nbWorkers); continue; }
-                if (longCommandWArg(&argument, "--memlimit")) { NEXT_UINT32(memLimit); continue; }
-                if (longCommandWArg(&argument, "--memory")) { NEXT_UINT32(memLimit); continue; }
+                if (longCommandWArg(&argument, "--memory") || longCommandWArg(&argument, "--memlimit")) {
```

Comment:
minor (for information): 
writing the branch this way makes the following (obviously wrong) command valid : 
`--memory=--memlimit=`
because `&argument` is updated twice.

Reply:
I'll fix this!